### PR TITLE
Implementation of the GameRules::CheckWinner() method[s].

### PR DESCRIPTION
**Pull request by JannikWell**
I just implemented the Methods:
-isValidCoord()
-checkWinner()
-checkHorizontalWinner()
-checkVerticalWinner()
-checkLeftToTopWinner()
-checkRightToTopWinner()
in the GameRules Class.

I also updated the class diagram file.

Furthermore, i had to set the method GameBoard::getTokenAt() to public.

### CODE REVIEW

File: FourInARow/FourInARowLib/GameRules.h [hunk 1]
```diff
@@ -13,10 +14,17 @@ class GameRules
     bool isValidMove(int column);
     bool placeToken(int column);
 
-    TeamEnum checkWinner();
+    bool checkWinner(TeamEnum team);
 
     std::shared_ptr<GameBoard> getGameBoard();
 
   private:
+    bool isValidCoord(int x, int y);
```
Comment (Bennik2000) [inline FourInARow/FourInARowLib/GameRules.h:22]: I implemented a isValidCoordinate in the GameBoard class. After both PRs have been merged, we should consider to call the GameBoard's function.

File: FourInARow/FourInARowLib/GameRules.cpp [hunk 2]
```diff
@@ -5,22 +5,179 @@
...
+    }
+
+    return false;
+}
+
+bool GameRules::checkLeftToTopDiagonalWinner(TeamEnum team) {
+    int counter = 0;
+    int diagonalX, diagonalY;
+
+    for (int x = 0; x < (gameBoard->getWidth() - 3); x++) {
```
Comment (Bennik2000) [inline FourInARow/FourInARowLib/GameRules.cpp:91]: -3 is a magic number. 
Would REQUIRED_FOR_WINNING - 1 be the same? If so I would consider using this.

File: FourInARow/FourInARowLib/GameRules.cpp [hunk 3]
```diff
@@ -5,22 +5,179 @@
...
+    }
+
+    return false;
+}
+
+bool GameRules::checkRightToTopDiagonalWinner(TeamEnum team) {
+    int counter = 0;
+    int diagonalX, diagonalY;
+
+    for (int x = (gameBoard->getWidth() - 1); x > 2; x--) {
```
Comment (Bennik2000) [inline FourInARow/FourInARowLib/GameRules.cpp:140]: `x > 2` same magic number as above.  I guess `x >= REQUIRED_FOR_WINNING` would be the same.

File: FourInARow/FourInARowLib/GameRules.cpp [hunk 4]
```diff
@@ -5,22 +5,179 @@
...
+            else {
+                counter = 0;
+            }
+
+            diagonalX--;
+            diagonalY--;
+        }
+    }
+
+    for (int y = (gameBoard->getHeight() - 2); y > 2; y--) {
```
Comment (Bennik2000) [inline FourInARow/FourInARowLib/GameRules.cpp:161]: `x > 2` and `gameBoard->getHeight() - 2` is the same magic number as above.

File: FourInARow/FourInARowLib/GameRules.h (hunk 1 above)
Comment (JannikWell) [reply FourInARow/FourInARowLib/GameRules.h:22]: Okay, for sure.

File: FourInARow/FourInARowLib/GameRules.cpp (hunk 4 above)
Comment (JannikWell) [reply FourInARow/FourInARowLib/GameRules.cpp:161]: Yes, sorry. I totally forgot about that 🤦 

I will fix that magic number problem asap.

**Pull request closed without merging**